client/state/data-layer/wpcom/sites/admin-menu/index.js

```diff
@@ -17,8 +18,38 @@ export const requestFetchAdminMenu = ( action ) =>
 		action
 	);
 
-export const handleSuccess = ( { siteId }, menuData ) => {
-	return receiveAdminMenu( siteId, menuData );
+const sanitizeUrl = ( url, site ) => {
+	if ( new RegExp( `^https?://${ site?.domain }` ).test( url ) ) {
+		return url;
+	} else if ( /^\//.test( url ) ) {
```

Comment:
It might be worth assigning this expression to a const with a descriptive name for future devs who may come across this code.

Reply:
Totally, amended in f851171